internal/backends/nodejs/grab.go

```diff
@@ -57,193 +54,122 @@ var internalModules = []string{
+func nodejsGuess() (map[api.PkgName]bool, bool) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		util.Die("couldn't get working directory: %s", err)
+	}
 
-func parseFile(contents []byte, results chan parseResult) {
-	language := javascript.GetLanguage()
+	dir := os.DirFS(cwd)
 
 	importsQuery := `
 (import_statement
-	source: (string) @import)
+  source: (string) @import)
 
-(
-	(call_expression
-		function: (identifier) @function
-		arguments: (arguments ((_) @other-imports)? ((string) @import) .))
-	(#eq? @function "require"))
+((call_expression
+     function: [(identifier) @function
+		            (import)]
+     arguments: (arguments . (string) @import .))
+ (#eq? @function "require"))
 `
 
-	query, err := sitter.NewQuery([]byte(importsQuery), language)
+	js := javascript.GetLanguage()
+	jsPkgs, err := util.GuessWithTreeSitter(dir, js, importsQuery, jsPathGlobs, []string{})
 	if err != nil {
-		// This should never happen - the query is hardcoded
-		log.Fatalln(err, importsQuery)
+		util.Die("couldn't guess imports: %s", err)
 	}
 
-	qc := sitter.NewQueryCursor()
+	ts := typescript.GetLanguage()
+	tsPkgs, err := util.GuessWithTreeSitter(dir, ts, importsQuery, tsPathGlobs, []string{})
+	if err != nil {
+		util.Die("couldn't guess imports: %s", err)
+	}
 
-	node, err := sitter.ParseCtx(context.Background(), contents, language)
+	tsx := tsx.GetLanguage()
+	tsxPkgs, err := util.GuessWithTreeSitter(dir, tsx, importsQuery, tsxPathGlobs, []string{})
 	if err != nil {
-		results <- parseResult{nil, false}
-		return
+		util.Die("couldn't guess imports: %s", err)
 	}
 
-	qc.Exec(query, node)
+	modules := append([]string{}, jsPkgs...)
+	modules = append(modules, tsPkgs...)
+	modules = append(modules, tsxPkgs...)
 
-	importPaths := []string{}
-	for {
-		match, ok := qc.NextMatch()
-		if !ok {
-			break
-		}
+	return findImports(modules), true
+}
 
-		match = qc.FilterPredicates(match, contents)
-		if len(match.Captures) == 0 {
+func findImports(names []string) map[api.PkgName]bool {
+	pkgs := map[api.PkgName]bool{}
+
+	for _, mod := range names {
+		if mod == "" {
 			continue
 		}
 
-		var importPath string
-
-		if match.PatternIndex == 0 {
-			importPath = match.Captures[0].Node.Content(contents)
-		} else if match.PatternIndex == 1 {
-			// TODO: https://github.com/smacker/go-tree-sitter/issues/110
-			// once the above issue is resolved, uncomment the `@other-imports` predicate
-			// and remove this check
-			if len(match.Captures) != 2 {
-				continue
-			}
-
-			importPath = match.Captures[1].Node.Content(contents)
+		if pkgs[api.PkgName(mod)] {
+			continue
 		}
 
-		// TODO: https://github.com/smacker/go-tree-sitter/issues/111
-		// once the above issue is resolved, use the string destructuring used in the
-		// `import_statement` pattern above in the `call_expression` pattern instead
-		// of using this hacky trim
-		importPaths = append(importPaths, strings.Trim(importPath, "'\"`"))
```

Comment:
still haven't ported over the quote stripping, and for some reason the `@org/package` import isn't working, but most of the output is correct :)